### ISSUE TITLE
Thanos-Receive cannot run without object storage configuration

### ISSUE THREAD

**Issue by itayvallach** [thanos-io/thanos#5171]
Thanos, Prometheus and Golang version used:

quay.io/thanos/thanos:v0.24.0

Object Storage Provider:
None

What happened:
When trying to run thanos receive without an object storage config, I get the following error:

flag objstore.config-file or objstore.config is required for running this command and content cannot be empty.
What you expected to happen:
Thanos runs and only uses the data-dir, without trying to write to external object storage.

Do I have to use a FILESYSTEM object storage target instead? Why do I also need to specify a --data-dir in that case?

**Comment by yeya24** [comment]
Yes, looks like right now a bucket is required and you can only use a FILESYSTEM bucket in this case.
`--data-dir` this is the local data directory for TSDB, where Thanos receiver keeps the latest 2h data before it uploads to the bucket.

To me, I think it is unnecessary to require an object storage for receiver. But there's some logic to change for sure

**Comment by itayvallach** [comment]
Still necessary for our use case (no object storage, data-dir only using PV/PVC).

**Comment by badrabubker** [comment]
@itayvallach: Are you using helm ? and if so which helm chart.
I believe the problem is not in Thanos but in the helm chart as I am facing the same problem

**Comment by badrabubker** [comment]
I was able to run thanos receiver binary without object store config
I submitted this issue https://github.com/bitnami/charts/issues/10995 for the bitnami helm chart
